### PR TITLE
Add principles to style guide

### PR DESCRIPTION
**Pull request by kathancox**
The style team has decided to include some principles for our style guide. The aim is to provide a framework for our decision process when writing, reviewing, and creating further style guidelines. These follow Cockroach Labs' values.

### CODE REVIEW

**Comment by kathancox** [comment]
Hey @rmloveland @ericharmeling I have added the principles to the top of the guide. I probably haven't introduced them in the most refined way—so would appreciate any thoughts there! :-) 

cc @mikeCRL @lnhsingh

**Review by rmloveland — APPROVED** [review]
LGTM!

File: StyleGuide.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
-CockroachDB docs should be:
+CockroachDB docs follow these principles:
 
-- Clear
-- Correct
-- Concise
+- **Commit to Excellence:** We commit to publishing documentation that serves our users and customers with a focus on excellence. We take pride in writing clear, concise, and correct docs, constantly iterating, and aiming to produce great work to help our users.
+- **Communicate Openly and Honestly:** We produce our best documentation when we communicate openly and honestly with our users. Understanding that documenting features and limitations transparently will enable users to effectively use Cockroach Labs products.
```
Comment (mikeCRL) [inline StyleGuide.md:4]:
```suggestion
- **Communicate Openly and Honestly:** We produce our best documentation when we communicate openly and honestly with our users. Documenting features and limitations transparently enables users to effectively use Cockroach Labs products while instilling confidence in the docs as a trusted resource.
```

File: StyleGuide.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
+- **Respect:** We aim to write humble, positive, friendly, and above all else helpful documentation. We appreciate every user through inclusive, accessible, and non-hyperbolic language.
+- **Establish balance:** We establish balance between complexity and accessibility for all users. We describe complex problems through accessible technical language and links to further information, without obfuscating meaning through unnecessarily complicated language. We also establish balance by always considering our style guidelines, but we break the rules when it’s better for the user or promotes our other values.
```
Comment (mikeCRL) [inline StyleGuide.md:6]:
```suggestion
- **Establish balance:** We establish balance between complexity and accessibility for all users. We describe complex problems through accessible technical language and links to further information, without obfuscating meaning through unnecessarily complicated language. We also establish balance by always considering our style guidelines, but we can break these rules—or propose new rules—when it’s better for the user or promotes our other values.
```

File: StyleGuide.md [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
 
 The following guidance is provided to ensure consistency.
```
Comment (mikeCRL) [inline StyleGuide.md:8]: While we're here

```suggestion
The following guidance is provided to benefit authors and reviewers by reflecting past style decisions while promoting consistency and readability across our content, which benefits all readers.
```

**Review by mikeCRL — CHANGES_REQUESTED** [review]
This is great. I left a few suggestions.

**Comment by kathancox** [comment]
Hey @mikeCRL should be there now!

**Review by mikeCRL — APPROVED** [review]
LGTM!

**Pull request merged**